File: tests/fixtures/typescript/policies/pii_redaction_before_llm/compliant/src/handlers.ts
import { redact } from './redactor';
import * as anthropic from 'anthropic';

export function summarizeUser(userId: string): string {
  const user = loadUser(userId);
  const safe = redact(user);
  return anthropic.messages.create({ model: 'claude', input: safe });
}

export function loadUser(userId: string): { id: string; email: string } {
  return { id: userId, email: 'x@example.com' };
}

File: tests/fixtures/python/policies/pii_redaction_before_llm/compliant/app/handlers.py
from app import redactor
import anthropic


def summarize_user(user_id: str) -> str:
    user = load_user(user_id)
    safe = redactor.redact(user)
    return anthropic.messages.create(model="claude", input=safe)


def load_user(user_id: str) -> dict:
    return {"id": user_id, "email": "x@example.com"}
